velox/dwio/parquet/tests/ParquetTestBase.h

```diff
@@ -126,16 +138,63 @@ class ParquetReaderTestBase : public testing::Test {
     auto rowReaderOpts = getReaderOpts(fileSchema);
     rowReaderOpts.setScanSpec(scanSpec);
     auto rowReader = reader->createRowReader(rowReaderOpts);
-    assertReadExpected(fileSchema, *rowReader, expected, *pool_);
+    assertReadExpected(fileSchema, *rowReader, expected, *leafPool_);
+  }
+
+  std::unique_ptr<dwio::common::FileSink> createSink(
+      const std::string& filePath) {
+    auto sink = dwio::common::FileSink::create(
+        fmt::format("file:{}", filePath), {.pool = rootPool_.get()});
+    EXPECT_TRUE(sink->isBuffered());
+    EXPECT_TRUE(fs::exists(filePath));
+    EXPECT_FALSE(sink->isClosed());
+    return sink;
+  }
+
+  std::unique_ptr<facebook::velox::parquet::Writer> createWriter(
+      std::unique_ptr<dwio::common::FileSink> sink,
+      std::function<std::unique_ptr<facebook::velox::parquet::DefaultFlushPolicy>()> flushPolicy,
+      facebook::velox::common::CompressionKind compressionKind) {
+    facebook::velox::parquet::WriterOptions options;
+    options.memoryPool = rootPool_.get();
+    options.flushPolicyFactory = flushPolicy;
+    options.compression = compressionKind;
+    return std::make_unique<facebook::velox::parquet::Writer>(std::move(sink), options);
+  }
+
+  std::pair<std::unique_ptr<facebook::velox::parquet::Writer>, dwio::common::FileSink*>
+  createWriterWithSinkPtr(
```

Comment:
This API is odd. I see that this is used by `SinkTest.cpp`. I think we should simply inline this function.

Reply:
This API will further also be used by the Parquet Writer Tests as well.

Reply:
There is already a `createSink` and `createWriter` API. Users can use them without much effort. Let's remove this API.
```
auto sink = createSink(filePath);
auto sinkPtr = sink.get();
auto writer = createWriter(std::move(sink), ...);
```

Reply:
Thanks @majetideepak. Made the necessary changes !